app/controllers/flow_controller.rb

```diff
@@ -10,18 +10,26 @@ def start
   end
 
   def show
-    @title = presenter.title
+    state = SmartAnswer::ResolveState.new(flow).from_response_store(response_store, node_name)
+    @presenter = FlowPresenter.new(flow, state)
+    @title = @presenter.title
 
-    if params[:node_slug] == presenter.node_slug
-      render presenter.current_node.view_template_path, formats: [:html]
+    if params[:node_slug] == @presenter.node_slug
+      render @presenter.current_node.view_template_path, formats: [:html]
     else
-      redirect_to flow_path(id: params[:id], node_slug: presenter.node_slug, params: forwarding_responses)
+      redirect_to flow_path(id: params[:id],
+                            node_slug: @presenter.node_slug,
+                            params: @presenter.forwarding_responses)
```

Comment:
They seem part of State, since they're part of the state of your journey through a flow. We have the advantage that keeping them there also means FlowPresenter doesn't have to be concerned about what type of response store is used. 

I've updated this code to pull the info from state since the object is in scope.